Atlas/backend/authentication/tests.py

```diff
@@ -25,16 +25,21 @@ def setUp(self):
             'password': self.password,
             'confirm_password' :self.password,
         }
+        self.login_data =  {
+            'username_or_email' : self.username,
+            'password'          : self.password,
+        }
+
 
         self.client = Client()
 
-    def test_jwt_login_json(self):
+    def test_jwt_login_json_with_username(self):
         """
-        Ensure JWT login view using JSON POST works.
+        Ensure JWT login view using JSON POST works with only username and password.
         """
         response = self.client.post(
             self.login_url,
-            json.dumps(self.data),
+            json.dumps(self.login_data),
```

Comment:
I think using login data with username globally, and login data with email locally is confusing. Why not dictate every test case to specify their login data themselves? We can do so by removing any login data available globally to each test method.

Reply:
Why do you think it is confusing ? I think having a global login data shortens the code a lot. Maybe it would be better to change its name as login_data_with_username, what do you think @yigitozkavci ?

Reply:
When writing tests, readability takes over writing less code, hence it's generally preferable to have everything there when reading tests. Even duplications are not so important in tests, and global variables makes them so hard to read, unless you share that variable across many test cases.

For instance, to understand this line: https://github.com/SaitTalhaNisanci/bounswe2017group10/blob/462d39b1773012867e708af90e147c39c758616a/Atlas/backend/authentication/tests.py#L78
I need to know the whole context of tests, just because of that 2 lines of shared code.

Why doesn't this apply to backend code other than tests? Because we need to maintain and refactor that code, hence its architecture. But for tests, we need to be able to read them so that we understand what application does at that part.

So in this case, `login_data_with_username` would make it obvious that you are referring to a valid login data. But for modifications like the one I mentioned above, I highly suggest not modifying global data in place, but rather create a new one.